### PR TITLE
Added base SubscriptionsAPI filters benchmarks

### LINKED ISSUE #3234: New Event Filtering: Benchmark Java implementation of Base SubscriptionsAPI filters
**Problem**
After #3233, we should add benchmarks for all of the base SubscriptionsAPI filters. 

To add a benchmark for a new type of filter, add a file to [this folder](https://github.com/knative-extensions/eventing-kafka-broker/tree/main/data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter) for the filter type, and then follow the pattern in https://github.com/knative-extensions/eventing-kafka-broker/pull/3247/files#diff-b300a22e800560bd785d1c686237dbd3bc39979e85ef6699adbbbec58e2101ca. Specifically, when making a new benchmark, you need to extend the `FilterBenchmark` class and override the `createEvent` and `createFilter` functions to create the desired event and filter to benchmark.

**Exit Criteria**
Working benchmarks for each of:

- [ ] Exact filter
- [ ] Not filter
- [ ] Prefix filter
- [ ] Suffix filter

**Time Estimate (optional):** 
How many developer-days do you think this may take to resolve? 1

**Additional context (optional)**
- The names of the benchmark classes created in your PR should be added to https://github.com/knative-extensions/eventing-kafka-broker/blob/main/data-plane/benchmarks/resources/filter-class-list.txt
- https://github.com/knative/eventing/issues/5204
- https://github.com/knative/eventing/issues/7111
- https://github.com/knative-extensions/eventing-kafka-broker/pull/3247

### PR DESCRIPTION
**Pull request by Cali0707**
Fixes #3234 

## Proposed Changes

- Add benchmarks for not filter
- Add benchmarks for prefix filter
- Add benchmarks for suffix filter

### CODE REVIEW

**Comment by Cali0707** [comment]
Here are the first two sets of benchmark results:
[ExactFilterBenchmark.1692720467.out.txt](https://github.com/knative-extensions/eventing-kafka-broker/files/12411210/ExactFilterBenchmark.1692720467.out.txt)
[NotFilterBenchmark.1692720467.out.txt](https://github.com/knative-extensions/eventing-kafka-broker/files/12411211/NotFilterBenchmark.1692720467.out.txt)

File: data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SuffixFilterBenchmark.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                .withDataContentType("testContentType")
+                .withSubject("testSubject")
+                .build();
+    }
+
+    public static class SuffixFilterIDBenchmark extends FilterBenchmark {
+
+        @Override
+        protected Filter createFilter() {
+            return new PrefixFilter(Map.of(CloudEventV1.ID, "lmnop"));
```
Comment (Leo6Leo) [inline data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SuffixFilterBenchmark.java:43]: Just curious if the prefix that need to be matched is really long, will that have impact on the benchmark result? @Cali0707

Comment (Leo6Leo) [reply data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SuffixFilterBenchmark.java:43]: And why it is `PrefixFilter` here?

File: data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SuffixFilterBenchmark.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        protected CloudEvent createEvent() {
+            return event();
+        }
+    }
+
+    public static class SuffixFilterAllContextAttributes3CharsBenchmark extends FilterBenchmark {
+
+        @Override
+        protected Filter createFilter() {
+            return new PrefixFilter(Map.of(
```
Comment (Leo6Leo) [inline data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SuffixFilterBenchmark.java:75]: same question: why it is Prefix Filter here?

File: data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/PrefixFilterBenchmark.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            return event();
+        }
+    }
+
+    public static class PrefixFilterLongNoBenchmark extends FilterBenchmark {
+
+        @Override
+        protected Filter createFilter() {
+            return new PrefixFilter(Map.of(
+                    CloudEventV1.ID, "qwertyuiopasdfghjklzxcvbnm",
```
Comment (Leo6Leo) [inline data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/PrefixFilterBenchmark.java:95]: Just curious if the prefix that need to be matched is really long, will that have impact on the benchmark result?

File: data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/NotFilterBenchmark.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    public static CloudEvent event() {
+        return CloudEventBuilder.v1()
+                .withId("abcdefghijklmnop")
+                .withSource(URI.create("http://localhost"))
+                .withType("com.github.pull.create")
+                .withDataSchema(URI.create("/api/schema"))
+                .withDataContentType("testContentType")
+                .withSubject("testSubject")
+                .build();
+    }
```
Comment (Leo6Leo) [inline data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/NotFilterBenchmark.java:40]: This part of the code is duplicated across all these 3 files. If these classes are in a shared package or module, consider abstracting this to a utility class or a common parent class maybe?

File: data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SampleEvent.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package dev.knative.eventing.kafka.broker.dispatcher.impl.filter;
```
Comment (Leo6Leo) [inline data-plane/benchmarks/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/impl/filter/SampleEvent.java:1]: boilerplate missing here

**Comment by Leo6Leo** [comment]
/lgtm
I don't see any other problems, great work @Cali0707 !!

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**